cloud_storage_clients/s3_client: tmp log skip

joe-redpanda opened the pull request:
Investigation is ongoing into '400 Bad Request' responses received by
s3_client when performing PUT operations. The 'error' log lines
generated by this issue are causing consistent bleed in CI investigation
time.

This PR adds these particular 400s to the default allowlist for bad logs
in non-cloud redpanda tests.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

rockwotj commented on src/v/cloud_storage_clients/s3_client.cc:
```diff
@@ -580,8 +580,11 @@ ss::future<result<T, error_outcome>> s3_client::send_request(
+            // TODO downgraded temporarily to stop CI bleed while
+            // investigation to S3 400s is ongoing
             vlog(
-              s3_log.error,
+              s3_log.warn,
```
Can we specifically only do `400` errors?
```suggestion
vlogv(
              err.status_code == 400 ? ss::log_level::warn : ss::log_level::error,
```

oleiman commented on src/v/cloud_storage_clients/s3_client.cc:
```diff
@@ -580,8 +580,11 @@ ss::future<result<T, error_outcome>> s3_client::send_request(
+            // investigation to S3 400s is ongoing
             vlog(
-              s3_log.error,
+              s3_log.warn,
```
Is there a reason to prefer this change over adding the problematic log line to the default allow lists in `redpanda.py`?

i don't feel too strongly about it, but i'd like to capture opinions on the PR

joe-redpanda replied:
That sounds like a much better path, I wasn't aware that was a feature

WillemKauf commented:
Yeah, if we're going to do this, better as a temporary change in ducktape allow lines rather than a change in core code.

https://github.com/redpanda-data/redpanda/blob/dc72932b03f2bbeadb89b0c629f5d9c2f8837775/tests/rptest/tests/random_node_operations_test.py#L37-L47

joe-redpanda commented:
Swapped to be part of the default bad logs ignore.

As written, this should only target this specific flavor of 400 given that the telltale symptom is
1. 400 bad request
2. no rid
3. fixed error message '400 Bad Request'

joe-redpanda committed:
cloud_storage_clients/s3_client: tmp bad log skip

Investigation is ongoing into '400 Bad Request' responses received by
s3_client when performing PUT operations. The 'error' log lines
generated by this issue are causing consistent bleed in CI investigation
time.

This PR adds these particular 400s to the default allowlist for bad logs
in non-cloud redpanda tests.

rockwotj commented on tests/rptest/services/redpanda.py:
```diff
@@ -144,6 +144,10 @@ class CloudStorageCleanupStrategy(enum.Enum):
     # just indicate a race in committing to Iceberg.
     re.compile(r"UpdateRequirement.*Assert"),
+
+    # Temporary: there is an ongoing investigation into s3_client receiving 400 Bad Request. For now, stop the CI bleed
```
consider linking to the JIRA, so folks know if/when this is bit rot

joe-redpanda replied:
ack, fixing

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--e325d6f4-e214-40cd-a88e-1040f0abae6c-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/69380">build#69380</a></summary>


| test_class              | test_method         | test_arguments            | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                    |
|:------------------------|:--------------------|:--------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------|
| MountUnmountIcebergTest | test_simple_remount | {"cloud_storage_type": 1} | ducktape    | https://buildkite.com/redpanda/redpanda/builds/69380#01982eae-14d7-404b-a85d-20381e1fe82d | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS |
</details>
<!--/body-->
